### PR TITLE
feat: button inactive until change

### PR DESCRIPTION
**Pull request by jogeshwar01**
Closes: #8738 

### Description:

Added code to keep save/update buttons inactive until a change is made. Also includes cancel button to revert back to original state.

- [x] The Save Button should be disabled until a change has been made
- [x]  Once a change has been made, there should be an option to cancel that change. When I click on this, the changes revert and the Save button goes back to being disabled

Areas of impact in the form settings
- [x] User settings - Profile form
- [x] User settings - Password form
- [x] Project - Tables Configuration form
- [x] Organizations - General form
- [ ]  Organizations - Embedding configuration form

Just wanted to check on the `Organizations - Embedding configuration form`.
I do see it in the docs - https://docs.lightdash.com/references/embedding/. Is the code for this closed-source for now?

Output:

Profile Form -
Before:

image

On Change:

image

Table Configuration Form-
Before:

image

On Change:

image

General Form-
Before:

image

On Change:

image

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

**Comment by PriPatel** [comment]
@jogeshwar01 
Screenshot 2024-01-30 at 18 51 01
Thanks for this! For consistency and usability, would it be possible to have the buttons like the example above where they are next to eachother ? (instead of at opposite ends?) 

Thanks again!!

**Comment by jogeshwar01** [comment]
@PriPatel Sure, will update the alignment.

**Comment by jogeshwar01** [comment]
@PriPatel Fixed the alignment and updated the output screenshots.

**Review by PriPatel — COMMENTED** [review]
Screenshots looks good, we're just going to test the behaviour in a mirror PR!

File: packages/frontend/src/components/UserSettings/PasswordPanel/index.tsx
Comment (IrakliJani) [inline packages/frontend/src/components/UserSettings/PasswordPanel/index.tsx:64]: hey @jogeshwar01 why do we need to set default value to the form and then resetDirty? can you explain how this works?

Comment (IrakliJani) [reply packages/frontend/src/components/UserSettings/PasswordPanel/index.tsx:64]: this seems to me that it's serving two purposes, one is initializing the data and another one is resetting `dirty` state. I think this should be called `handleFormReset` and should be called from cancel button only. LMK!

File: packages/frontend/src/components/ProjectTablesConfiguration/ProjectTablesConfiguration.tsx
Comment (IrakliJani) [inline packages/frontend/src/components/ProjectTablesConfiguration/ProjectTablesConfiguration.tsx:170]: this seems repetitive. can we have a one object and use that to call both of the functions? more questions about those two functions below

**Review by IrakliJani — CHANGES_REQUESTED** [review]
thanks for the contribution @jogeshwar01 I added few questions in the PR. please, have a look and let me know! thanks 🙌

File: packages/frontend/src/components/ProjectTablesConfiguration/ProjectTablesConfiguration.tsx
Comment (jogeshwar01) [reply packages/frontend/src/components/ProjectTablesConfiguration/ProjectTablesConfiguration.tsx:170]: @IrakliJani  Thanks for the review, will update this to remove the repetition.

File: packages/frontend/src/components/UserSettings/PasswordPanel/index.tsx
Comment (jogeshwar01) [reply packages/frontend/src/components/UserSettings/PasswordPanel/index.tsx:64]: @IrakliJani The `form.setValues` is used to track the current values in the form while the `form.resetDirty` controls our current state that also helps us to display/hide the Cancel button on change

For instance, once a user clicks on `Save changes/Update` , we would want to set both our form values as well as the dirty state to the current changed state (to prevent Cancel button from showing after state has been updated)

So basically, we need this to be called in 2 cases - either user Saves the changes (useEffect) or cancels them.

**Comment by jogeshwar01** [comment]
@IrakliJani Fixed the redundancy in the code and changed the function name to `handleFormReset` for `PasswordPanel`.

File: packages/frontend/src/components/UserSettings/PasswordPanel/index.tsx
Comment (jogeshwar01) [reply packages/frontend/src/components/UserSettings/PasswordPanel/index.tsx]: In case of Password Panel, as we redirect to /login when we update the password, we call it from the Cancel button only.
There are no dependencies for the `handleFormReset useCallback` and so it would never go inside the useEffect except  for the first time.

**Comment by IrakliJani** [comment]
hi @jogeshwar01

I've made few additional changes to this PR:
- upgraded to mantine form v7 which introduces `setInitialValues` and simplifies the logic
- use ZOD for validation in all of the forms

waiting for e2e tests before I merge this. thanks for your work 🙌

**Review by IrakliJani — APPROVED** [review]

**Pull request merged**
